src/sql/workbench/browser/modelComponents/listView.component.ts

```diff
@@ -18,6 +19,15 @@ import { IListRenderer, IListVirtualDelegate } from 'vs/base/browser/ui/list/lis
 import { attachListStyler } from 'vs/platform/theme/common/styler';
 import { ScrollbarVisibility } from 'vs/base/common/scrollable';
 import { ILogService } from 'vs/platform/log/common/log';
+import { createIconCssClass } from 'sql/workbench/browser/modelComponents/iconUtils';
+
+interface ListViewOption {
+	id: string;
+	label: string;
+	ariaLabel?: string;
+	icon?: azdata.IconPath;
+	addIconPadding?: boolean;
```

Comment:
this can be removed, right?

Reply:
I still need to add a flag to list item template to add the additional padding.

Reply:
you are using the icon information to decide the value of addIconPadding, which is redundant.

Reply:
If in case only some options have icons, the icons would be undefined for the other options. This will just make sure that the List option renderer is forced to add icon padding.

Reply:
my point is: it is a calculated field, you shouldn't need to expose it in the interface:
![image](https://user-images.githubusercontent.com/13777222/186722669-8d56408e-62db-4f9f-8b5c-168b62ff1289.png)

Reply:
This is not exposed to the extensions and is only local to the component. I have removed this property from azdata.proposed.ts.

Reply:
yes, I know, but the places that might use this component still see this property and will be confused, you should have an internal flag somewhere to track whether icon padding should be added for the list.

Reply:
if it is not very straightforward to implement, please make this one optional as well as add comment explaining the why it is here and can't be removed.